datadog_checks_base/datadog_checks/base/checks/base.py

```diff
@@ -402,25 +404,50 @@ def in_developer_mode(self):
         self._log_deprecation('in_developer_mode')
         return False
 
+    def find_typos_in_options(self, user_config, models_config, level):
+        user_config = user_config or {}
+        models_config = models_config or {}
+        typos = []  # type: List[str]
+
+        known_options = dict(models_config).keys()
+        unknown_options = sorted(list(user_config.keys() - known_options))
+        for unknown_option in unknown_options:
+            most_similar_known_option = (None, 0)
+            for known_option in known_options:
+                ratio = jaro_distance(unknown_option, known_option)
+                if ratio > most_similar_known_option[1]:
+                    most_similar_known_option = (known_option, ratio)
```

Comment:
This is an unlikely scenario, but should we have the option of multiple `most_similar_known_option`s?

Reply:
True, there could be a few candidate known options. We could either only show all if they have an equal score (when rounded), or show all that are above the similarity threshold.

Reply:
I like the idea of showing all candidates that are above the similarity threshold since it can give you more options, but maybe the threshold would need to be higher to avoid too many candidates?

Reply:
Updated to include multiple candidates!